### ISSUE TITLE
Table Recognition model training overfitting and long training time

### ISSUE THREAD

**Issue by johntoma** [PaddlePaddle/PaddleOCR#9443]
Hello, after training, I realised my model is overfitting as the accuracy is very good on the images I used to train but quite bad on other images.

I am training over the `ch_ppstructure_mobile_v2.0_SLANet`, using `SLANet_ch.yml` config file.

I have a custom dataset I created of table images which I generated using TableGeneration, there are `5000 training images` and `500 validation images`.

When I evaluate the model, I get an accuracy of 95%, but when I actually test the model on table images that wasnt used in my training dataset, it tends to have trouble predicting accurately. It's a bit strange.

What might be causing this? is my `epoch_num` too high? It is set at 400 (default). Should I simply increase the number of training images, if so, by how much? I'm thinking of increasing my training data size, but I will probably need to adjust the hyperparameter values differently and I'm not sure how to determine what the right values are.

In the config file, I have set `learn_empty_box` to `True`, as I would like for the model to learn how to detect empty cells very accurately. I think this might be affecting the results too, maybe 5000 images is probably not enough if I want to set this value to True.

My batch_size is set to 32 (due to hardware limitations), I also only have 1 GPU to use, and the training took about 4.5 days, I'm not sure if this is a reasonable time or if my config file settings have made the training take longer than it needs to.

Below is my config file. Can I please get some ideas of which hyperparameters to adjust?  Thanks

```
Global:
  use_gpu: True
  epoch_num: 400
  log_smooth_window: 20
  print_batch_step: 20
  save_model_dir: ./output/SLANet_ch
  save_epoch_step: 400
  # evaluation is run every 331 iterations after the 0th iteration
  eval_batch_step: [0, 331]
  cal_metric_during_train: True
  pretrained_model: models/ch_ppstructure_mobile_v2.0_SLANet_train/best_accuracy
  checkpoints: 
  save_inference_dir: ./output/SLANet_ch/infer
  use_visualdl: False
  infer_img: ppstructure/docs/table/table.jpg
  # for data or label process
  character_dict_path: ppocr/utils/dict/table_structure_dict_ch.txt
  character_type: en
  max_text_length: &max_text_length 500
  box_format: &box_format xyxyxyxy # 'xywh', 'xyxy', 'xyxyxyxy'
  infer_mode: False
  use_sync_bn: True
  save_res_path: output/infer

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  clip_norm: 5.0
  lr:
    learning_rate: 0.00017
  regularizer:
    name: 'L2'
    factor: 0.00000

Architecture:
  model_type: table
  algorithm: SLANet
  Backbone:
    name: PPLCNet
    scale: 1.0
    pretrained: True
    use_ssld: True
  Neck:
    name: CSPPAN
    out_channels: 96
  Head:
    name: SLAHead
    hidden_size: 256
    max_text_length: *max_text_length
    loc_reg_num: &loc_reg_num 8

Loss:
  name: SLALoss
  structure_weight: 1.0
  loc_weight: 2.0
  loc_loss: smooth_l1

PostProcess:
  name: TableLabelDecode
  merge_no_span_structure: &merge_no_span_structure True

Metric:
  name: TableMetric
  main_indicator: acc
  compute_bbox_metric: False
  loc_reg_num: *loc_reg_num
  box_format: *box_format
  del_thead_tbody: True

Train:
  dataset:
    name: PubTabDataSet
    data_dir: train_data/table/train/
    label_file_list: [train_data/table/train/train.txt]
    transforms:
      - DecodeImage:
          img_mode: BGR
          channel_first: False
      - TableLabelEncode:
          learn_empty_box: True
          merge_no_span_structure: *merge_no_span_structure
          replace_empty_cell_token: False
          loc_reg_num: *loc_reg_num
          max_text_length: *max_text_length
      - TableBoxEncode:
          in_box_format: *box_format
          out_box_format: *box_format
      - ResizeTableImage:
          max_len: 488
      - NormalizeImage:
          scale: 1./255.
          mean: [0.485, 0.456, 0.406]
          std: [0.229, 0.224, 0.225]
          order: 'hwc'
      - PaddingTableImage:
          size: [488, 488]
      - ToCHWImage:
      - KeepKeys:
          keep_keys: [ 'image', 'structure', 'bboxes', 'bbox_masks', 'shape' ]
  loader:
    shuffle: True
    batch_size_per_card: 32
    drop_last: True
    num_workers: 1

Eval:
  dataset:
    name: PubTabDataSet
    data_dir: train_data/table/val/
    label_file_list: [train_data/table/val/val.txt]
    transforms:
      - DecodeImage:
          img_mode: BGR
          channel_first: False
      - TableLabelEncode:
          learn_empty_box: True
          merge_no_span_structure: *merge_no_span_structure
          replace_empty_cell_token: False
          loc_reg_num: *loc_reg_num
          max_text_length: *max_text_length
      - TableBoxEncode:
          in_box_format: *box_format
          out_box_format: *box_format
      - ResizeTableImage:
          max_len: 488
      - NormalizeImage:
          scale: 1./255.
          mean: [0.485, 0.456, 0.406]
          std: [0.229, 0.224, 0.225]
          order: 'hwc'
      - PaddingTableImage:
          size: [488, 488]
      - ToCHWImage:
      - KeepKeys:
          keep_keys: [ 'image', 'structure', 'bboxes', 'bbox_masks', 'shape' ]
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: 32
    num_workers: 1

```

**Comment by johntoma** [comment]
How can I calculate what the right number of epocs would be? If I have 20,000 training images for example, how many epochs should I have?

**Comment by andyjpaddle** [comment]
1. Dataset generated by using TableGeneration is usually simple and is easy to overfitting, you should add some real dataset.
2. If you use pretrained model to finetune, you can decrease epochs and lr, for 20000 images, 50epoch is enough

**Comment by johntoma** [comment]
@andyjpaddle thanks for the reply. 

If 20000 images is 50 epochs, would that mean every 10k images should be 25 epochs? 

You also mentioned to decrease lr, for me its currently at 0.00017 with batch size of 32. Is this fine, or if not, what should I decrease it to?

**Comment by andyjpaddle** [comment]
the epoch is a round number, you can try it, lr is depends on  batch size, for 32, you can try 1e-4 or 5e-5 in finetune training.
Also, you can do some ablation.

**Comment by johntoma** [comment]
@andyjpaddle Thanks again for your reply. 
I just have one more question.

I've increased my training set to 25000 imgs and validation set to 2500 imgs. I'm trying an epoch_num of 60 with a learning rate of 0.0001 (1e-4). 

My eval_batch_step is [0, 331], I think I should probably change this value to something different, but I'm not sure to what. Do you have any idea of what I should change this to?

Thanks!

**Comment by andyjpaddle** [comment]
eval_batch_step is mainly for choosing best model, you can compute a rough value according to the iters in each epoch, usually you can eval once in each epoch.

**Comment by johntoma** [comment]
@andyjpaddle ok, thanks a lot for your help!
